src/common/providers/https.ts

```diff
@@ -500,6 +504,55 @@ interface CallableTokenStatus {
   auth: TokenStatus;
 }
 
+function decodeToken(token: string): unknown {
+  if (!JWT_REGEX.test(token)) {
+    return {};
+  }
+  const components = token
+    .split('.')
+    .map((s) => Buffer.from(s, 'base64').toString());
+  let payload = components[1];
+  if (typeof payload === 'string') {
+    try {
+      const obj = JSON.parse(payload);
+      if (typeof obj === 'object') {
+        payload = obj;
+      }
+    } catch (e) {}
+  }
+  return payload;
+}
+
+/**
+ * Decode, but not verify, a Auth ID token.
+ *
+ * Do not use in production. Token should always be verified using the Admin SDK.
+ *
+ * This is exposed only for testing.
```

Comment:
And for debug mode/emulator, right?

Reply:
I think the comment is correct - this function will only be called internally. It will be executed in debug mode but will never be invoked directly.